### DISCUSSION TITLE
How to set `max_form_parts`?

### DISCUSSION THREAD

**Discussion by jessica369tt** [pallets/werkzeug#2636] [Q&A]
How do I set the `max_form_parts` attribute? I am using Flask and want to set it to 5000.

**Comment by davidism** [comment]
Generally, you should increase it for specific requests only, so that only requests that require a higher number allow it.

```python
from flask import request

@app.post("/complex")
def complex_post():
    request.max_form_parts = 5000
    ... access request.form
```

You can use a subclass instead to set it for a specific application:

```python
from flask.wrappers import Request

class MyRequest(Request):
    max_form_parts = 5000

app.request_class = MyRequest
```

To set it globally, set the base class's attribute.

```python
from werkzeug.wrappers import Request

Reqeust.max_form_parts = 5000
```
